src/libraries/Microsoft.Extensions.Hosting.Abstractions/src/BackgroundService.cs

```diff
@@ -39,19 +39,15 @@ public abstract class BackgroundService : IHostedService, IDisposable
         /// <returns>A <see cref="Task"/> that represents the asynchronous Start operation.</returns>
         public virtual Task StartAsync(CancellationToken cancellationToken)
         {
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
What's the reason for adding this? If we do want this check, it should instead be:
```C#
if (cancellationToken.IsCancellationRequested)
{
    return Task.FromCanceled(cancellationToken);
}
```

Reply:
I wanted to preserve the behavior of throwing synchronously when calling the method with an already cancelled token.

> if we do want this check, it should instead be

Interesting, the reason I thought `ThrowIfCancellationRequested` was acceptable was because you previously said this https://github.com/dotnet/runtime/pull/116283#discussion_r2131453586.  I agree though that I need Task.FromCanceled if I want the same exception as before.  I'll adjust it.

Reply:
My comment isn't about the type of the exception but rather whether the exception propagates out of the method synchronously or asynchronously. We generally consider it a bug if non-argument exceptions propagate out of a Task-returning method synchronously.

Reply:
I guess this is in a similar boat as above - there's a race here where we check and throw - however that I'm willing to accept that race since it's necessary to capture the case where there was no race at all (already canceled token).